bwapi/include/BWAPI/BroodwarOutputDevice.h

```diff
@@ -71,25 +67,19 @@ namespace BWAPI
         if (c != '\n')
         {
           buffer.sputc(c);
-          bufferWidth += charWidth(c);
+          bufferWidth += charPrintingWidth(c);
         }
       }
       return n;
     }
 
   private:
-    int charWidth(char c) const
+    int charPrintingWidth(char c) const
     {
-      const BW::Font& font = *BW::BWDATA::FontBase[Text::Size::Default];
-      int mw = font.maxWidth();
-      if (c == ' ')
-        return mw / 2;
-      else if (c == '\t')
-        return mw * 2 - bufferWidth % (mw * 2);
-      else if (font.getChar(c))
-        return font.getChar(c)->width() + 1;
+      if (c == '\t')
+        return DEFAULT_FONT_MAX_WIDTH * 2 - bufferWidth % (DEFAULT_FONT_MAX_WIDTH * 2);
       else
-        return 0;
+        return DEFAULT_FONT_CHAR_PRINTING_WIDTHS[c];
```

Comment:
You'll want to possibly cast the char to unsigned, otherwise this will underflow for character values > 127.